projects/packages/blaze/src/class-dashboard-rest-controller.php

```diff
@@ -709,6 +729,54 @@ public function edit_dsp_generic( $path, $req, $args = array() ) {
 		);
 	}
 
+	/**
+	 * Will check the posts for prices and add them to the posts array
+	 *
+	 * @param WP_REST_Request $posts The posts object.
+	 * @return array|WP_Error
+	 */
+	protected function add_prices_in_posts( $posts ) {
+
+		if ( ! function_exists( 'wc_get_product' ) ||
+			! function_exists( 'wc_get_price_decimal_separator' ) ||
+			! function_exists( 'wc_get_price_thousand_separator' ) ||
+			! function_exists( 'wc_get_price_decimals' ) ||
+			! function_exists( 'get_woocommerce_price_format' ) ||
+			! function_exists( 'get_woocommerce_currency_symbol' )
+		) {
+			return $posts;
+		}
+
+		foreach ( $posts as $key => $item ) {
+			if ( ! isset( $item['ID'] ) ) {
+				$posts[ $key ]['price'] = '';
+				continue;
+			}
+			$product = wc_get_product( $item['ID'] );
+			if ( $product ) {
```

Comment:
You may have to be more defensive here, just like in #35566, to avoid fatals in some scenarios.

Reply:
@jeherve check out this comment by Panos

https://github.com/Automattic/jetpack/pull/35066#discussion_r1488101621

i think we should be ok

Reply:
I'm afraid that's not going to be enough. In some scenarios, Woo extensions may change what's returned there. That's something we ran into a few times in the past, and why we ended up checking for `$product instanceof WC_Product` in the PR I linked to earlier. Also see #11301 for another example.

Reply:
Ok fixed, added the extra condition

Reply:
Being extra defensive doesn't hurt :)